js/components/portal-dashboard/all-responses-popup/popup-student-response-list.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+
+import css from "../../../../css/portal-dashboard/all-responses-popup/popup-student-response-list.less";
+
+interface IProps {
+  students: any; // TODO: add type
+  isAnonymous: boolean;
+}
+
+export class PopupStudentResponseList extends React.PureComponent<IProps> {
+  render() {
+    const { students, isAnonymous } = this.props;
+    return (
+      <div className={css.responseTable} data-cy="popup-response-table">
+        {students && students.map((student: any, i: number) => {
+          const formattedName = isAnonymous
+            ? student.get("name")
+            : `${student.get("lastName")}, ${student.get("firstName")}`;
+
+          return (
```

Comment:
I was a little thrown off by seeing this return inside of the render return.  Might be clearer if this is refactored into a function.  renderStudentRow or something similar.

Reply:
Also Code Climate has a point here.  Maybe we should make a utility function that takes a student name and returns the formatted name since we're now doing this in multiple places.

Reply:
Where we added `question-utils` and `answer-utils`, we could add `student-utils` and start with this function:
```
export const getFormattedStudentName = (isAnonymous: boolean, student: Map<any, any>) => {
  return (
    isAnonymous
      ? student.get("name")
      : `${student.get("lastName")}, ${student.get("firstName")}`
  );
};
```

Then we could call the utility function in both `StudentNames` and `PopupStudentResponseList`:
`const formattedName = getFormattedStudentName(isAnonymous, student);`

What do you think?

Reply:
Was thinking the same thing but didn't know where to put it. Will give it a try.

Reply:
I would just make a new module called `student-utils` and put it next to the other two mentioned above.  Unless you see a better location.

Reply:
Done. Added a util/student-utils.ts which has the above function. Replaced it in both the student-names and popup-student-response-list